### PR TITLE
fix CI failures on macos

### PR DESCRIPTION
**Pull request by alessandrod**
This fixes the current CI failures we're seeing on macos https://github.com/aya-rs/aya/actions/runs/8810405556/job/24182725423

The fix is to make sure bpf-linker is always rebuilt/relinked against the latest llvm installed via brew.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -258,7 +258,7 @@ jobs:
         if: runner.os == 'macOS'
         # NB: rustc doesn't ship libLLVM.so on macOS, so disable proxying (default feature).
         # Remove --rev when LLVM18 is released.
```
Comment (tamird) [inline .github/workflows/ci.yml:260]: looks like llvm is 18 now: https://formulae.brew.sh/formula/llvm

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -258,7 +258,7 @@ jobs:
...
-        run: cargo install bpf-linker --git https://github.com/aya-rs/bpf-linker.git --rev 821f92990074cb7e950e25129dcd55e20424cede --no-default-features
+        run: cargo install --force bpf-linker --git https://github.com/aya-rs/bpf-linker.git --rev 821f92990074cb7e950e25129dcd55e20424cede --no-default-features
```
Comment (tamird) [inline .github/workflows/ci.yml:261]: can you add a comment to explain why we're doing this?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alessandrod) [reply .github/workflows/ci.yml]: removed

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alessandrod) [reply .github/workflows/ci.yml]: done

**Review by tamird — APPROVED** [review]

**Pull request merged**